src/crypto/crypto_basic.cpp

```diff
@@ -104,8 +104,8 @@ t_hash_PRV Hash1_PRV( const t_hash_PRV & hash ) {
     const unsigned char* u_hashmsg;
     u_hashmsg = reinterpret_cast<const unsigned char *>( hash.c_str() ); // read locked
 
-   	const size_t out_u_hash_len = 64; // Hash1_size();
-   	assert( out_u_hash_len == Hash1_size() );  // <-- ^-- TODO(rob) constexpr instead?
+	constexpr size_t out_u_hash_len = 64; // Hash1_size();
+	assert( out_u_hash_len == Hash1_size() && "Hash1 size assert [64]");
 
```

Comment:
Both const and constexpr should be ok, and should work with static_assert().

TODO: try to static_assert this.

The trick with && "assert message" in assert is fine I guess.